File: binarySearch.cpp
#include<iostream>
#include<stdio.h>
using namespace std;
int bsearch(int x[],int n,int no)
{
    int l,u,m;
    l=0;
    u=n-1;
    while(l<=u)
     {
         m=(l+u)/2;
         if(x[m]==no)
           return m+1;
         else
           if(x[m]<no)
             l=m+1;
           else 
             u=m-1;    
     }return -1;
}
int main()
{
    int x[10]={1,2,3,4,5,6,7,8,9,10};
    int no;
    cout<<"Enter the index of no you want"<<endl;
    cin>>no;
    int res=bsearch(x,10,no);
    cout<<"Index number is ="<<endl;
    cout<<res;
   return 0; 
}

File: binary_search.java
import java.util.*;

public class binary_search {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the length of the array : ");
        int n = sc.nextInt();
        int[] array = new int[n];
        for (int i = 0; i < n; i++) {
            System.out.println("Enter the value at " + i + "th index :");
            array[i] = sc.nextInt();
        }
        Arrays.sort(array);
//        for (int i=0;i<n ; i++){
//            System.out.print(array[i]+"\t");
//        }
        System.out.println("Enter the number you want to find : ");
        int test= sc.nextInt();
        int low=0;
        int high=n-1;
        int mid = 0;
        boolean a = false;
        while(low <= high){
            mid=(low+high)/2;
            if(array[mid]==test){
                a=true;
                break;
            }
            else if (array[mid]<test) {
                low = mid+1;
            }
            else {
                high=mid-1;
            }
        }
        if (a==true){
            System.out.println("The number "+test+" is in the index "+mid);
        }
        else {
            System.out.println("The number is not present in the array");
        }
    }
}